custom-script/utils.py

```diff
@@ -14,7 +18,38 @@
 MODEL_INPUTS_STR = ", ".join(MODEL_INPUTS)
 
 
-def parse_subtree(node, brackets=True):
+def prepare_data(eopatches: np.ndarray, train_eopatches: np.ndarray) -> Tuple[np.ndarray]:
+    FEATURES_SAMPLED = FeatureType.DATA, "FEATURES_SAMPLED"
+    IS_DATA_SAMPLED = FeatureType.MASK, "IS_DATA_SAMPLED"
+    LABELS_SAMPLED = FeatureType.MASK_TIMELESS, "water_label_SAMPLED"
+
+    # Set the features and the labels for train and test sets
+    features_train = np.array([eopatch[FEATURES_SAMPLED] for eopatch in eopatches[train_eopatches]])
+    labels_train = np.array([eopatch[LABELS_SAMPLED] for eopatch in eopatches[train_eopatches]])
+    mask_train = np.array([eopatch[IS_DATA_SAMPLED] for eopatch in eopatches[train_eopatches]])
+
+    features_test = np.array([eopatch[FEATURES_SAMPLED] for eopatch in eopatches[~train_eopatches]])
+    labels_test = np.array([eopatch[LABELS_SAMPLED] for eopatch in eopatches[~train_eopatches]])
+    mask_test = np.array([eopatch[IS_DATA_SAMPLED] for eopatch in eopatches[~train_eopatches]])
+
+    # get shape
+    p1, t, w, h, f = features_train.shape
+    p2, t, w, h, f = features_test.shape
+
+    # reshape to n x m
+    features_train = np.moveaxis(features_train, 1, 3).reshape(p1 * w * h, t * f)
+    labels_train = np.moveaxis(labels_train, 1, 2).reshape(p1 * w * h, 1).squeeze()
+    mask_train = np.moveaxis(mask_train, 1, 2).reshape(p1 * w * h, 1).squeeze()
+
+    features_test = np.moveaxis(features_test, 1, 3).reshape(p2 * w * h, t * f)
+    labels_test = np.moveaxis(labels_test, 1, 2).reshape(p2 * w * h, 1).squeeze()
+    mask_test = np.moveaxis(mask_test, 1, 2).reshape(p2 * w * h, 1).squeeze()
+
+    # remove points with no valid data
+    return features_train[mask_train], labels_train[mask_train], features_test[mask_test], labels_test[mask_test]
+
+
+def parse_subtree(node: Dict, brackets: bool = True) -> str:
```

Comment:
use `dict` if you're not providing any further types, so it doesn't need to be imported

Reply:
done